Include/pythread.h

```diff
@@ -127,7 +127,7 @@ struct _Py_tss_t {
 #undef NATIVE_TSS_KEY_T
 
 /* When static allocation, you must initialize with Py_tss_NEEDS_INIT. */
-#define Py_tss_NEEDS_INIT   {0}
+#define Py_tss_NEEDS_INIT   ((Py_tss_t) {0})
```

Comment:
Sadly, The compound literals is not compatible with C++ standard :persevere: (it's a C99 feature) 
A similar issue of the C-API header pointed out at related https://github.com/python/cpython/pull/1362#discussion_r136357005.

Reply:
Ok, then ... should we just distinguish the C and C++ cases? And maybe even MSVC from the real compilers, as usual?

I mean, this is a public header file. It's fairly annoying if it constrains the applicability of its definitions. I guess I'll have to open a new ticket for this one after all...

Reply:
FYI, we've solved an initialization issue of a TSS key by using temporary variables. https://github.com/python/cpython/pull/1362#discussion_r137957323
https://github.com/python/cpython/blob/731e18901484c75b60167a06a0ba0719a6d4827d/Python/pystate.c#L49-L54